performance_manager/lib/logging_utils.py

```diff
@@ -18,70 +17,75 @@ def __init__(self, process_name: str, **metadata: MdValues) -> None:
         create a process logger with a name and optional metadata. a start time
         and uuid will be created for timing and unique identification
         """
-        self.parent = "performance_manager"
-        self.process_name = process_name
-        self.metadata = metadata
-        self.uuid: Optional[uuid.UUID] = None
+        self.default_data: Dict[str, Any] = {}
+        self.metadata: Dict[str, Any] = {}
+
+        self.default_data["parent"] = "performance_manager"
+        self.default_data["process_name"] = process_name
+
+        # default_data keys that can not be added as metadata
+        self.protected_keys = [
+            "parent",
+            "process_name",
+            "process_id",
+            "uuid",
+            "status",
+            "duration",
+            "error_type",
+        ]
+
         self.start_time = 0.0
-        self.process_id = 0
+
+        self.add_metadata(**metadata)
+
+    def _get_log_string(self) -> str:
+        """create logging string for log write"""
+        logging_list = []
+        # add default data to log output
+        for (key, value) in self.default_data.items():
+            logging_list.append(f"{key}={value}")
+
+        # add metadata to log output
+        for (key, value) in self.metadata.items():
+            logging_list.append(f"{key}={value}")
+
+        return ", ".join(logging_list)
 
     def add_metadata(self, **metadata: MdValues) -> None:
         """add metadata to the process logger"""
         for key, value in metadata.items():
-            self.metadata[key] = value
+            # skip metadata key if protected as default_data key
+            # maybe raise on this? instead of fail silently
+            if key in self.protected_keys:
+                continue
+            self.metadata[str(key)] = str(value)
 
     def log_start(self) -> None:
         """log the start of a proccess"""
-        self.uuid = uuid.uuid4()
+        self.default_data["uuid"] = uuid.uuid4()
+        self.default_data["process_id"] = os.getpid()
+        self.default_data["status"] = "started"
+        self.default_data.pop("duration", None)
+        self.default_data.pop("error_type", None)
```

Comment:
initiate log with required `default_data` if `duration` or `error_type` already in dictionary, get rid of them